modules/nrfxlib/nrf_802154/sl/platform/nrf_802154_platform_sl_lptimer_zephyr.c

```diff
@@ -201,3 +256,103 @@ uint64_t nrf_802154_platform_sl_lptimer_sync_lpticks_get(void)
 {
 	return m_sync_timer.target_time;
 }
+
+nrf_802154_sl_lptimer_platform_result_t nrf_802154_platform_sl_lptimer_hw_task_prepare(
+	uint64_t fire_lpticks,
+	uint32_t ppi_channel)
+{
+	bool done_on_time;
+	uint32_t evt_address;
+	nrf_802154_sl_mcu_critical_state_t mcu_cs_state;
+
+	if (!hw_task_state_set(HW_TASK_STATE_IDLE, HW_TASK_STATE_SETTING_UP)) {
+		/* The only one available set of peripherals is already used. */
+		return NRF_802154_SL_LPTIMER_PLATFORM_NO_RESOURCES;
+	}
+
+	/* Allocate the channel if not already done. */
+	if (m_hw_task.chan == RTC_CHAN_INVALID) {
+		/* The channel allocation cannot take place during module
+		 * initialization, because for nRF53 it would run
+		 * out of resources - some other module temporarily needs
+		 * an rtc channel to initialize. For this reason, this is where
+		 * the allocation is made.
+		 * Once a channel has been successfully allocated, it will be held
+		 * until nrf_802154_platform_sl_lp_timer_deinit is called.
+		 */
+		m_hw_task.chan = z_nrf_rtc_timer_chan_alloc();
+		if (m_hw_task.chan < 0) {
+			m_hw_task.chan = RTC_CHAN_INVALID;
+			hw_task_state_set(HW_TASK_STATE_SETTING_UP, HW_TASK_STATE_IDLE);
+			return NRF_802154_SL_LPTIMER_PLATFORM_NO_RESOURCES;
+		}
+	}
+
+	if (z_nrf_rtc_timer_set(m_hw_task.chan, fire_lpticks, NULL, NULL) != 0) {
+		hw_task_state_set(HW_TASK_STATE_SETTING_UP, HW_TASK_STATE_IDLE);
+		return NRF_802154_SL_LPTIMER_PLATFORM_TOO_DISTANT;
+	}
+
+	evt_address = z_nrf_rtc_timer_compare_evt_address_get(m_hw_task.chan);
+
+	nrf_802154_sl_mcu_critical_enter(mcu_cs_state);
+
+	/* For triggering to take place a safe margin is 2 lpticks from `now`. */
+	if ((z_nrf_rtc_timer_read() + 2) > fire_lpticks) {
+		/* it is too late */
+		nrf_802154_sl_mcu_critical_exit(mcu_cs_state);
+		cc_unbind(m_hw_task.chan, ppi_channel);
+		m_hw_task.ppi = NRF_802154_SL_HW_TASK_PPI_INVALID;
+		z_nrf_rtc_timer_abort(m_hw_task.chan);
+		done_on_time = false;
```

Comment:
Why is `done_on_time` needed?
Can't we simply do `hw_task_state_set(HW_TASK_STATE_SETTING_UP, HW_TASK_STATE_IDLE);` 
and then 
`return NRF_802154_SL_LPTIMER_PLATFORM_TOO_LATE;` - this is done like this with all previous errors, I don't see a reason why this should be done differently